scripts/extract-api.ts

```diff
@@ -0,0 +1,40 @@
+import * as fs from 'fs-extra';
```

Comment:
should we just add this as a pre-push hook?
We do something like this for split-e2e-tests, which generates our circleci template & verifies correctness

Reply:
@awsluja I have explained that in PR description.

Reply:
we may also be able to leverage lerna/nx so that this script runs only for packages that have changed

Reply:
@awsluja I added verification on push and harnessed nx to make it more preformant.